src/Common/ThreadPool.cpp

```diff
@@ -272,22 +278,40 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
 
         if (!need_shutdown)
         {
+            // set up tracing context for this thread by its parent context
+            DB::OpenTelemetryThreadTraceContextScope thread_trace_context("ThreadPool::worker()" ,
+                                                                          parent_thead_trace_context);
+
             try
             {
                 ALLOW_ALLOCATIONS_IN_SCOPE;
                 CurrentMetrics::Increment metric_active_threads(
                     std::is_same_v<Thread, std::thread> ? CurrentMetrics::GlobalThreadActive : CurrentMetrics::LocalThreadActive);
 
                 job();
+
+                if (thread_trace_context.root_span.isTraceEnabled())
+                {
+                    // Use the thread name as operation name so that the tracing log will be more clear.
+                    // the thread name is usually set in the jobs, we can only get the name after the job finishes
+                    std::string thread_name = getThreadName();
+                    if (!thread_name.empty())
+                        thread_trace_context.root_span.operation_name = thread_name;
+                }
+
                 /// job should be reset before decrementing scheduled_jobs to
                 /// ensure that the Job destroyed before wait() returns.
                 job = {};
+                parent_thead_trace_context.reset();
```

Comment:
Maybe I miss something, but I don't really see why the parent thread trace ctx must be reset all (also not in l. 314).

Reply:
Just make sure the weak reference of system.opentelemetry_span_log is released. Or it will be released until current thread is scheduled to execute another task.